examples/vmlinux_header/vmlinux_505_structs.h

```diff
@@ -0,0 +1,181 @@
+#ifndef __OLD_KERNEL_STRUCTS_H__
+#define __OLD_KERNEL_STRUCTS_H__
+
+struct task_struct_505 {
+	struct thread_info thread_info;
+	volatile long int state;
+	void *stack;
+	refcount_t usage;
+	unsigned int flags;
+	unsigned int ptrace;
+	struct llist_node wake_entry;
+	int on_cpu;
+	unsigned int cpu;
```

Comment:
you don't need all the fields of task_struct, only few necessary fields can be defined. It's also better to leave such definitions close to BPF source code that's using this type. See https://nakryiko.com/posts/bpf-core-reference-guide/#handling-incompatible-field-and-type-changes for some more examples and details.

Reply:
>  Also, I was expecting that search path to vmlinux.h would need to be changed somewhere either in build script or somewhere else. Don't see it here. What am I missing?

Not sure what you mean by this.  I've moved vmlinux_514.h into the examples directory, and softlinked a vmlinux.h to it.  I then softlink each examples vmlinux.h to the example/vmlinux.h file.  The build scripts do not need to be aware of the vmlinux.h file.

```
pwd && ls -l 
/code/single-vmlinux/examples/runqslower/src/bpf
total 8
-rw-r--r-- 1 mm mm 2677 Nov 10 20:46 runqslower.bpf.c
-rw-r--r-- 1 mm mm  232 Nov  8 19:00 runqslower.h
lrwxrwxrwx 1 mm mm   18 Nov 10 19:15 vmlinux.h -> ../../../vmlinux.h
```

Reply:
Never mind, I forgot we do vmlinux.h symlinking in each example to avoid needing to specify extra -I<path> to the compiler. Please keep vmlinux_514.h name and leave all the vmlinux.h as symlinks

Reply:
> you don't need all the fields of task_struct, only few necessary fields can be defined.

@anakryiko I didn't know you could do this neither do I know the concept behind this. The closes thing I know is in OOPs where you can assign child object to variable of type parent and just refer to the fields the parent knows about. Is this only possible in bpf CO-RE code or can you do this in regular C as well? I would love to read about this concept if its possible in regular C.

Reply:
It's BPF CO-RE thing. Only fields that are accessed are necessary (to let compiler generate CO-RE relocation that will be used by libbpf to find corresponding type and field in the full kernel BTF definition).